diff --git a/stdlib/public/runtime/Metadata.cpp b/stdlib/public/runtime/Metadata.cpp
index 2e466715ffd..f6685108742 100644
--- a/stdlib/public/runtime/Metadata.cpp
+++ b/stdlib/public/runtime/Metadata.cpp
@@ -757,7 +757,7 @@ MetadataResponse swift::swift_getCanonicalSpecializedMetadata(
   if (auto *token =
           description
               ->getCanonicalMetadataPrespecializationCachingOnceToken()) {
-    cacheCanonicalSpecializedMetadata(description, token);
+    //cacheCanonicalSpecializedMetadata(description, token);
     // NOTE: If there is no token, then there are no canonical prespecialized
     //       metadata records, either.
   }
@@ -802,7 +802,7 @@ MetadataResponse swift::swift_getCanonicalPrespecializedGenericMetadata(
     const TypeContextDescriptor *description, swift_once_t *token) {
   description = swift_auth_data_non_address(
       description, SpecialPointerAuthDiscriminators::TypeDescriptor);
-  cacheCanonicalSpecializedMetadata(description, token);
+  //cacheCanonicalSpecializedMetadata(description, token);
 
   return _swift_getGenericMetadata(request, arguments, description);
 }
