diff --git a/unittests/runtime/CMakeLists.txt b/unittests/runtime/CMakeLists.txt
index 60f9944e1e6..4ef65667c88 100644
--- a/unittests/runtime/CMakeLists.txt
+++ b/unittests/runtime/CMakeLists.txt
@@ -51,6 +51,8 @@ if(("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "${SWIFT_PRIMARY_VARIANT_SDK}") AND
       )
   elseif(SWIFT_HOST_VARIANT STREQUAL windows)
     list(APPEND PLATFORM_TARGET_LINK_LIBRARIES DbgHelp)
+  elseif(SWIFT_HOST_VARIANT STREQUAL linux)
+    list(APPEND PLATFORM_TARGET_LINK_LIBRARIES atomic)
   endif()
 
   if(SWIFT_ENABLE_EXPERIMENTAL_CONCURRENCY)
