diff --git a/unittests/runtime/LongTests/CMakeLists.txt b/unittests/runtime/LongTests/CMakeLists.txt
index 685d29f7547..386e41e1499 100644
--- a/unittests/runtime/LongTests/CMakeLists.txt
+++ b/unittests/runtime/LongTests/CMakeLists.txt
@@ -26,6 +26,8 @@ if(("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "${SWIFT_PRIMARY_VARIANT_SDK}") AND
       )
   elseif(SWIFT_HOST_VARIANT STREQUAL windows)
     list(APPEND PLATFORM_TARGET_LINK_LIBRARIES DbgHelp)
+  elseif(SWIFT_HOST_VARIANT STREQUAL linux)
+    list(APPEND PLATFORM_TARGET_LINK_LIBRARIES atomic)
   endif()
 
   add_swift_unittest(SwiftRuntimeLongTests
